### PR TITLE
feat(linting): add commit linter

### PR DESCRIPTION
**Pull request by stevennevins**
I'll repost this in Slack, but the steps to have this automatically work in your local environment are as follows:

 ```bash
 npm install

 npx husky install 

 npx husky add .husky/commit-msg  'npx --no -- commitlint --edit ${1}'
 
```

When you git commit -m "some message," husky will run as a hook on your commit message and validate that it is the appropriate format.  Additionally, it will run in the CI and verify that all commit messages in a PR and a single push are valid.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## Requirements
+
+Foundry
+Git
+Node.js
+
+## Setup Repo 
+
+```bash
+git clone --recurse-submodules -j8 git@github.com:sablier-labs/v2-core.git
```
Comment (ChaoticWalrus) [inline CONTRIBUTING.md:11]: is this the right command ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Commits must be linted
+
+A descriptive summary of the PR has been provided.
+
+### Environment Variables
+
+Some of the tests and features of this repository require environment variables to be set up.
+
+Copy the .env.example file to create a .env and populate it with the appropriate environment values that you have control control
```
Comment (ChaoticWalrus) [inline CONTRIBUTING.md:38]: typo: 'control control' => 'control of'

**Review by ChaoticWalrus — APPROVED** [review]
Looks good; we'll see if I regret the decision after this yells at me for commits 🙃 
Noted a couple minor things in the documentation to address before merging.

File: CONTRIBUTING.md (hunk 1 above)
Comment (stevennevins) [reply CONTRIBUTING.md:11]: rip

**Pull request merged**